### ISSUE TITLE
Allow using annotated tags as ref when pusing code

### ISSUE THREAD

**Issue by hsablonniere** [isomorphic-git/isomorphic-git#520]
Right now, if you use an annotated tag as a ref like this: `push({ ref: '1.0.0', remoteRef: 'master' })`, it will fail.

In [the code](https://github.com/isomorphic-git/isomorphic-git/blob/master/src/commands/push.js#L67) of the `push` command, there is a line where we `resolve` the ref to an oid:

```
let oid = await GitRefManager.resolve({ fs, gitdir, ref: fullRef, peel: true })
```

If the `fullRef` is an annotated tag, the `oid` will be the object id of the tag object itself and not the associated commit SHA1.

A bit like I proposed in #519, I would like `push()` to peel annotated tags when retrieving their commit SHA1. If we expose a `peelTags` option on the `expandRef()` command, we'll have to implement the option on the internal `GitRefManager.resolve()` which means it would be accessible to the push command.

WDYT?

**Comment by wmhilton** [comment]
Good catch! Will fix this weekend.

**Comment by wmhilton** [comment]
This was fixed in #522 and I just forgot to close this.
